src/App.tsx

```diff
@@ -181,8 +181,11 @@ class App extends React.Component<RouteComponentProps, State> {
   }
 
   public getContests = (ballotStyle: BallotStyle, election?: Election) =>
-    (election || this.state.election!).contests.filter(c =>
-      ballotStyle.districts.includes(c.districtId)
+    (election || this.state.election!).contests.filter(
+      c =>
+        ballotStyle.districts.includes(c.districtId) &&
+        ((!ballotStyle.partyId && !c.partyId) ||
+          ballotStyle.partyId === c.partyId)
```

Comment:
I feel like it took me to long to read/understand this logic. 

This reads clearer to me, what do you think?

```diff
-        ((!ballotStyle.partyId && !c.partyId) ||
-          ballotStyle.partyId === c.partyId)
+        (ballotStyle.partyId ? ballotStyle.partyId === c.partyId : !c.partyId)
```

Reply:
I have trouble parsing your version :)

My goal here is to say: if either one has a partyId, then they have to match.

Wait! Maybe this means we can make it as simple as this:

```
        ballotStyle.partyId === c.partyId
```

I think this works! If they're both undefined, it matches, if they're both defined, they have to be equal, and if only one of them is undefined, it's false, as expected.

Reply:
Nice! I just tried this and it works:

```js
      contest =>
        ballotStyle.districts.includes(contest.districtId) &&
        ballotStyle.partyId === contest.partyId
```

Reply:
FWIW…

```js
(ballotStyle.partyId ? ballotStyle.partyId === c.partyId : !c.partyId)
```

If `ballotStyle.partyId` is truthy  
`?` return true if `ballotStyle.partyId` matches `contest.partyId`,  
`:` otherwise return true if `contest.partyId` is false.